node/src/accountant/big_int_processing/big_int_divider.rs

```diff
@@ -0,0 +1,579 @@
+    }
+
+    #[test]
+    fn deconstruct_works_for_small_negative_number() {
+        let tested_number = -1;
+        let result = BigIntDivider::deconstruct(tested_number);
+
+        assert_eq!(result, (-1, i64::MAX));
+
+        assert_reconstitution(result, tested_number)
+    }
+
+    #[test]
+    fn deconstruct_works_for_big_negative_number() {
+        let tested_number = i64::MIN as i128;
+        let result = BigIntDivider::deconstruct(tested_number);
+
+        assert_eq!(result, (-1, 0));
+
+        assert_reconstitution(result, tested_number)
+    }
+
+    #[test]
+    fn deconstruct_and_reconstitute_works_for_number_just_slightly_smaller_than_the_low_b_type_size(
+    ) {
+        let tested_number = i64::MIN as i128 - 1;
+        let result = BigIntDivider::deconstruct(tested_number);
+
+        assert_eq!(result, (-2, 9223372036854775807));
+
+        assert_reconstitution(result, tested_number)
+    }
+
+    #[test]
+    fn deconstruct_works_for_huge_negative_number() {
+        let tested_number = 0xC0000000000000000000000000000000u128 as i128;
+        let result = BigIntDivider::deconstruct(tested_number);
+
+        assert_eq!(result, (-9223372036854775808, 0));
+
+        assert_reconstitution(result, tested_number)
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "Dividing big integer for special database storage: 0x40000000000000000000000000000000 is too big, maximally 0x3FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF allowed"
+    )]
+    fn deconstruct_has_its_limits_up() {
+        let _ = BigIntDivider::deconstruct(0x3FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF + 1);
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "Dividing big integer for special database storage: 0xBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF is too small, minimally 0xC0000000000000000000000000000000 allowed"
+    )]
+    fn deconstruct_has_its_limits_down() {
+        let _ = BigIntDivider::deconstruct((0xC0000000000000000000000000000000u128 as i128) - 1);
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "Reconstituting big integer from special database storage: the second, lower integer 0xFFFFFFFFFFFFFFFF is signed despite the requirement to be all-time positive"
+    )]
+    fn reconstitute_should_reject_lower_half_with_high_bit_set() {
+        let _ = BigIntDivider::reconstitute(0, -1);
+    }
+
+    #[test]
+    fn divided_integers_can_be_ordered() {
+        let a = i64::MAX as i128 * 23;
+        let b = i64::MAX as i128 + 1;
+        let c = i64::MAX as i128;
+        let d = (i64::MAX - 1) as i128;
+        let e = 5432;
+        let f = 0;
+        let g = -4567;
+        let h = (i64::MIN + 1) as i128;
+        let i = i64::MIN as i128;
+        let j = i64::MIN as i128 - 1;
+        let k = i64::MIN as i128 * 32;
+        let vec = vec![b, c, d, e, f, g, h, i, j, k];
+
+        let _ = vec.into_iter().enumerate().fold(
+            a,
+            |previous, current: (usize, i128)| {
```

Comment:
You could name the two components of `current` here, which would make this a bit easier to read.